File: backend/src/main/java/edu/cibertec/stockwarehouse/cabecerainventario/domain/model/CabeceraInventario.java
package edu.cibertec.stockwarehouse.cabecerainventario.domain.model;

import edu.cibertec.stockwarehouse.empleado.domain.model.Empleado;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tb_cabecerainventario")
public class CabeceraInventario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_cab_inventario")
    private int id_cab_inventario;

    @Column(name = "tipo_inventario")
    private int tipo_inventario;

    @Column(name = "fecha_cab_inventario", nullable = false)
    @Temporal(value = TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date fecha_cab_inventario;

    @ManyToOne
    @JoinColumn(name = "id_empleado")
    private Empleado empleado;

    @Column(name = "estado")
    private int estado;


    //método para asignar nombre a los estados
    public String NombreEstado() {
        String nombreEstado = null;
        switch (estado) {
            case 0:
                nombreEstado = "Activo";
                break;
            case 1:
                nombreEstado = "Anulado";
                break;
            default:
                nombreEstado = "";
        }
        return nombreEstado;
    }
}

File: frontend/src/app/cabecerainventario/modelos/CabeceraInventario.ts


export class CabeceraInventario {

    id_cab_inventario:number ;
    tipo_inventario:number;
    fecha_cab_inventario: Date ;
    estado:number;
    empleado:any;


   constructor(){
    this.id_cab_inventario = 0;
    this.tipo_inventario = 0;
    this.fecha_cab_inventario = new Date();
    this.estado = 0;
    this.empleado = {};
        
    }

    //metodo para asignar nombre a los tipos de inventario

    NombreInventario(tipo_inventario:number):string {

        let nombreInventario:string = "";

        switch(tipo_inventario){
            case 0:
                nombreInventario = "Entrada";
                break;
            case 1:
                nombreInventario = "Salida";
                break;
        }
        return nombreInventario;
    }

    // método para asignar nombre a los estados
 NombreEstado(estado:number):string {

    let nombreEstado:string = "";

   switch (estado) {
       case 0:
           nombreEstado = "Realizado";
           break;
       case 1:
           nombreEstado = "Pendiente";
           break;
       case 2:
           nombreEstado = "Anulado";
           break;
   }

   return nombreEstado;

}
}